util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java

```diff
@@ -87,15 +88,16 @@ public static void extract(ZipFile zipFile, File destDir)
 	 * @throws IOException
 	 *         if the entry could not be processed
 	 */
-	public static void writeEntry(ZipFile zipFile, ZipEntry entry, File destDir)
-		throws IOException
-	{
+	public static void writeEntry(ZipFile zipFile, ZipEntry entry, File destDir) throws IOException {
 		File outFile = new File(destDir, entry.getName());
 
+		if (! outFile.getCanonicalFile().toPath().startsWith(destDir.toPath())) {
```

Comment:
Looks like a decent way of checking for escape attempts. Btw. shouldn’t both paths be compared as canonical?

Reply:
good catch, added